src/server/db_slice.cc

```diff
@@ -1451,6 +1451,15 @@ void DbSlice::ResetEvents() {
   events_ = {};
 }
 
+void DbSlice::SetNotifyKeyspaceEvents(std::string_view notify_keyspace_events) {
+  if (notify_keyspace_events.empty()) {
+    for (auto& db : db_arr_) {
+      db->expired_keys_events_.clear();
+    }
+  }
+  expired_keys_events_recording_ = !notify_keyspace_events.empty();
```

Comment:
Good questions! `config set` is not transactional so why would this be an exception? Sure you might argue that for certain transactions there might be events that are "lost" but I would say this is fine + the only solution for this would be to make it a `global transaction` which I don't think it would be a great idea :man_shrugging: 

As for discarding for all previous events that's something I should check with Valkey. I will get back to you for this.

Reply:
@dranikpg I did check how it's implemented in `Valkey`. 

1. It does not clear the current keyspace events. So I will address this in a second and push. 
2. For transactions, yes it's "racy" if the keyspace events are disabled. So for certain transactions they might not include all the keyspace events for expirations but I would say this is fine. `CONFIG SET` is not transactional so :shrug:

Reply:
Let me know if you have any other thoughts

Reply:
Ok, if it's racy, let it be racy for us as well 🤷🏻 🙂